### PR TITLE
Batch of unit tests for `ConductorServerEndpoint`

### PR DESCRIPTION
**Pull request by smiclea**
Based on https://github.com/cloudbase/coriolis/pull/250.
Please merge #250 first.

### CODE REVIEW

**Review by Dany9966 — CHANGES_REQUESTED** [review]
Mostly LGTM

File: coriolis/tests/conductor/rpc/data/advance_execution_state.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        on_error: True
+        expected_status: PENDING
+
+# All parent tasks finalized and at least one non-error parent was completed
+# test 8
+- config:
+    tasks:
+      - id: task_1
+        status: CANCELED
+      - id: task_1
```
[the remark this answers is not in the document]
Comment (smiclea) [reply coriolis/tests/conductor/rpc/data/advance_execution_state.yml]: It's a typo. Should have been a different task id. Fixed now.

File: coriolis/tests/conductor/rpc/test_server.py [hunk 2]
```diff
@@ -1839,3 +1841,1264 @@ def create_task_side_effect(
...
+            task.task_type = task_type
+            call_handle_post_task_actions()
+            mock_update_minion_machine.assert_called_once_with(
+                mock.sentinel.context,
+                task_info['origin_minion_machine_id'],
+                {'provider_properties': [
+                    {'minion_provider_properties': 'value'}]},
+            )
+            mock_update_minion_machine.reset_mock()
+
```
[the remark this answers is not in the document]
Comment (smiclea) [reply coriolis/tests/conductor/rpc/test_server.py:2795]: nice catch!

**Comment by smiclea** [comment]
@Dany9966  All requested changes have been implemented. Thank you!

**Pull request merged**